nova/local_test.go

```diff
@@ -271,14 +271,28 @@ func (s *localLiveSuite) TestRunServerAvailabilityZoneNotAvailable(c *gc.C) {
 	c.Assert(err, gc.ErrorMatches, "(.|\n)*The requested availability zone is not available(.|\n)*")
 }
 
-func (s *localLiveSuite) TestAttachVolume(c *gc.C) {
+func (s *localLiveSuite) TestVolumeAttachments(c *gc.C) {
 
 	instance, err := s.createInstance("test-instance")
 	c.Assert(err, gc.IsNil)
 
+	// Test attaching a volume.
 	volAttachment, err := s.nova.AttachVolume(instance.Id, "volume-id", "/dev/sda1")
 	c.Assert(err, gc.IsNil)
-
 	c.Check(volAttachment.ServerId, gc.Equals, instance.Id)
 	c.Check(volAttachment.VolumeId, gc.Equals, "volume-id")
+
+	// Test listing volumes.
```

Comment:
If they were unit tests, I would agree. As integration tests, I think they cover a nice cross-section of the code without duplication of effort.